breacharbiter.go

```diff
@@ -141,40 +147,59 @@ func (b *breachArbiter) Start() error {
 
 	brarLog.Tracef("Starting breach arbiter")
 
-	// We load all pending retributions from the database and
-	// deterministically reconstruct a channel close summary for each. In
-	// the event that a channel is still open after being breached, we can
-	// use the close summary to reinitiate a channel close so that the
-	// breach is reflected in channeldb.
+	// TODO(roasbeef): instead use closure height of channel
+	_, currentHeight, err := b.cfg.ChainIO.GetBestBlock()
+	if err != nil {
+		return err
+	}
+
+	// Load all retributions currently persisted in the retribution store.
 	breachRetInfos := make(map[wire.OutPoint]retributionInfo)
-	closeSummaries := make(map[wire.OutPoint]channeldb.ChannelCloseSummary)
-	err := b.cfg.Store.ForAll(func(ret *retributionInfo) error {
-		// Extract emitted retribution information.
+	err = b.cfg.Store.ForAll(func(ret *retributionInfo) error {
 		breachRetInfos[ret.chanPoint] = *ret
-
-		// Deterministically reconstruct channel close summary from
-		// persisted retribution information and record in breach close
-		// summaries map under the corresponding channel point.
-		closeSummary := channeldb.ChannelCloseSummary{
-			ChanPoint:      ret.chanPoint,
-			ClosingTXID:    ret.commitHash,
-			RemotePub:      ret.remoteIdentity,
-			Capacity:       ret.capacity,
-			SettledBalance: ret.settledBalance,
-			CloseType:      channeldb.BreachClose,
-			IsPending:      true,
-		}
-		closeSummaries[ret.chanPoint] = closeSummary
-
 		return nil
 	})
 	if err != nil {
 		return err
 	}
 
+	// Load all currently closed channels from disk, we will use the
+	// channels that have been marked fully closed to filter the retribution
+	// information loaded from disk. This is necessary in the event that the
+	// channel was marked fully closed, but was not removed from the
```

Comment:
how can this happen?

Reply:
If we fail just before getting to this [line](https://github.com/lightningnetwork/lnd/blob/518778c9ea2c75c9faa9b30809947baf005738d9/breacharbiter.go#L632), this will catch it on restart